src/python/pants/backend/cc/lint/clangformat/subsystem.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+from typing import Iterable
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.option.option_types import SkipOption
+from pants.util.strutil import softwrap
+
+
+class ClangFormat(PythonToolBase):
+    options_scope = "clang-format"
+    name = "ClangFormat"
+    help = softwrap(
+        """
+        The clang-format utility for formatting C/C++ (and others) code
+        (https://clang.llvm.org/docs/ClangFormat.html).
+        """
+    )
+
+    default_version = "clang-format==14.0.3"
+    default_main = ConsoleScript("clang-format")
+
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.7"]
+
```

Comment:
Is registering lockfiles a thing if it's a Python tool? I saw a bunch of boilerplate around that recently in the Python linters, so didn't know if it was automated and being overridden - or if I should add it here?